industrial_ci/scripts/run_travis

```diff
@@ -0,0 +1,169 @@
+    def test_file(dir): # test if one of the file names is present
+        for n in  names:
+            p = os.path.join(dir, n)
+            if os.path.isfile(p):
+                return os.path.abspath(p)
+        return None
+
+    if len(args) >  0: # test first argument if available
+        p = test_file(args[0])
+        if p is not None:
+            return p, args[1:]
+    return test_file(os.getcwd()), args
+
+# parse range tuples from arguments
+def parse_ranges(args, offset=0):
+    r = []
+    def apply_offset(i): # adjust for user offsets
+        ao = int(i) + offset
+        if ao < 0:
+            raise ValueError("%s is not in supported range" % str(i))
+        return ao
+
+    for a in args:
+        if '-' in a: # range string
+            p1, p2 = a.split('-', 2)
+            if p1 == '' and len(r) == 0: # -X
+                p1 = 0
+            else: # X-[Y]
+                p1 = apply_offset(p1)
+            if p2 == '': # [X]-
+                r.append((p1, -1))
+                break
+            r.append((p1, apply_offset(p2)+1)) # X-Y
+        else:
+            i = apply_offset(a)
+            r.append((i, i+1))
+    return r
+
+def apply_ranges(ranges, num):
+    for start,end  in ranges:
+        if end == -1:
+            end = num
+        for i in range(start,end):
+            yield i
+
+
+def read_yaml(p):
+    with open(p) as f:
+        return yaml.safe_load(f)
+
+# read global and job-specific envs from
+def read_env(env):
+    m = env
+    g = ''
+    if isinstance(env, dict):
+        m = env['matrix']
+        if 'globals' in env:
+            g = ' '.join(env['globals']),
+    return g, m
+
+def parse_extra_args(args):
+    try:
+        extra = args.index('--')
+        return args[0:extra], args[extra+1:]
+    except ValueError:
+        return args, []
+
+env_assigment = re.compile(r"[a-zA-Z][a-zA-Z0-9_]*=")
+def gen_env(e):
+    if env_assigment.match(e):
+        return e
+    return '%s=%s' % (e, os.getenv(e, ''))
+
+
+def print_help(cmd):
+    print("""
+Usage: %s [PATH] [RANGE*] [-- [ENV*]]
+
+Parses the travis config in the given path or in the current working directory and runs all specified tests sequentially
+If no range is given, the list of jobs will get printed.
+
+The number of tests can be reduced by specifying one or more ranges:
+* single job: 1 (only first)
+* range: 2-3 (only second and third)
+* open start, only as first range: -4 (jobs 1 to 4)
+* open end, only as last range: 7- (job 7 and all following jobs)
+* open range: - (all jobs)
+
+Complex examples for a matrix wih 12 jobs:
+
+* -4 7 8: runs jobs 1 2 3 4 7 8
+* 1 7-9: runs jobs 1 7 8 9
+* 1 7-9 11-: runs jobs 1 7 8 9 11 12
+* -: runs all jobs
+
+The jobs will be run in clean environments.
+Only DOCKER_PORT, SSH_AUTH_SOCK, and TERM will be kept.
+Additional variable names can be passed at the end.
+
+""" % cmd)
+
+def main(scripts_dir, argv):
+    if '--help' in argv:
+        print_help(argv[0])
```

Comment:
Any reason you didn't use [argparse](https://docs.python.org/3/library/argparse.html) @ipa-mdl?

Reply:
It bloats the code and is overkill for just printing the usage.